### ISSUE TITLE
A complete XML example

### ISSUE THREAD

**Issue by Sarge2013** [mysqludf/lib_mysqludf_xml#2]
Dear Arnold,

I spent few hours with lib_mysqludf_xml - no doubts, it's very smart and powerful.
Please, could you help the newbies like me to add to your examples one simple BUT complete one - so that we could SEE HOW IT WORKS based on simple classic XML.
As to me - I'd be glad to pay for your time :) or beer :) - but to see how the functions produces something complete.

As I see it - such example starts from test 'yourtable' creation

---

```
CREATE TABLE `yourtable` (
`id` int(11) NOT NULL auto_increment,
`title` varchar(15), 
`text` varchar(255), 
`freenumber` varchar (15), 
PRIMARY KEY  (`id`)
) ENGINE=MyISAM DEFAULT CHARSET=utf8 DEFAULT COLLATE utf8_bin;

INSERT INTO yourtable
    (`id`, `title`, `text`, `freenumber`)
VALUES
    (1, 'Arnold', 'has created', '15'),
    (2, 'Daniels', 'this smart lib_mysqludf_xml', '1.5'),
    (3, 'Now', 'you can see', '10.52'),
    (4, 'How', 'this simple', '3'),
    (5, 'Example', 'will be transformed to XML', '123.587')
;
```

---

HERE YOU could give example with the lib_mysqludf_xml functions using
title AS title,
text AS text,
'freenumber' AS 'value' with a standartized decimal number .00 (just cut, no runaround) 
to have a final XML like below :
(including two lines below if possible - because not shown in the code due to an autoencoding on Github)
<Output xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

---

<?xml version="1.0" encoding="UTF-8"?>
<Output xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

```
<row>
    <title>Arnold</title>
    <text>has created</text>
    <value>15.00</value>
</row>
<row>
    <title>Daniels</title>
    <text>this smart lib_mysqludf_xml</text>
    <value>1.50</value>
</row>
<row>
    <title>Now</title>
    <text>you can see</text>
    <value>10.52</value>
</row>
<row>
    <title>How</title>
    <text>this simple</text>
    <value>3.00</value>
</row>
<row>
    <title>Example</title>
    <text>will be transformed to XML</text>
    <value>123.58</value>
</row>
```

</Output>

---

Thank you VERY MUCH IN ADVANCE for such simple but complete illustrative example !
Again, I'm sure any newbie will be thankful if such one will be included to the /test where is your "example_list",

Best regards,
Serge

**Comment by jasny** [comment]
Yeah the examples got lost :frowning:. Have a look at https://github.com/mysqludf/lib_mysqludf_xml/blob/master/API.

```
SELECT xql_agg('row' AS `xql:child`, `title`, `text`, `freenumber` AS `value`) FROM `yourtable`;
```

**Comment by Sarge2013** [comment]
Yeah :) this is exactly what I mean, dear Arnold.
So many interesting functions - but all spread - you should be a very good prof to understand how to compile a one complete solution.
By the way, sorry, your example above does not work. You could create table as above and try.
First it says "no column 'test' in field list" - when corrected to "title" - it says "no column 'row' in field list.
Sorry my joke - but it seems this is not easy even for you to compile the functions in a right way to have the output XML exactly as above. 
This is why I'm offering to write and show very simple but comprehensive example.
Simplest table as above -> your functions = simplest but finalized XML file.
After that everybody could read API and try to complicate basing on own needs.
It's just my opinion, you decide.
Have a nice day !
